fem/lininteg_boundary_flux.cpp

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "fem.hpp"
+#include "../fem/kernels.hpp"
+#include "../general/forall.hpp"
+
+namespace mfem
+{
+
+template<int T_D1D = 0, int T_Q1D = 0> static
+void BFLFEvalAssemble2D(const int nbe, const int d, const int q,
+                        const int *markers, const double *b,
+                        const double *weights, const Vector &coeff, double *y)
+{
+   const auto F = coeff.Read();
+   const auto M = Reshape(markers, nbe);
+   const auto B = Reshape(b, q, d);
+   const auto W = Reshape(weights, q);
+   const bool const_coeff = coeff.Size() == 1;
+   const auto C = const_coeff ? Reshape(F,1,1) : Reshape(F,q,nbe);
+   auto Y = Reshape(y, d, nbe);
+
+   MFEM_FORALL(e, nbe,
+   {
+      if (M(e) == 0) { return; } // ignore
```

Comment:
Should it be `continue` rather than `return`?

Reply:
The body in `MFEM_FORALL` is really lambda, so `return` behaves like `continue` (confusing, I know)

Reply:
A short comment can be helpful as a reminder: "'return' in lambda is like 'continue' in 'for' loop".

Reply:
Good idea. Done in 8ddbd5111.